### PR TITLE
new package: cns-nospec

### PR DESCRIPTION
**Pull request by jiseung**

### CODE REVIEW

File: var/spack/repos/builtin/packages/cns-nospec/package.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    the compressible Navier-Stokes equations. The code uses
+    8th order finite differences in space and a 3rd order,
+    low-storage TVD RK algorithm in time."""
+
+    homepage = "https://ccse.lbl.gov/ExaCT/index.html"
+    url      = "https://ccse.lbl.gov/ExaCT/CNS_Nospec.tgz"
+
+    version('master', '14ff5be62539d829b30b17281688ee3f')
+
+    variant('mpi', default=False, description='Build with MPI support')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cns-nospec/package.py]: This should default to True to match every other package in Spack.

File: var/spack/repos/builtin/packages/cns-nospec/package.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    variant('debug', default=True, description='Build with debugging')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cns-nospec/package.py]: This should default to False to match every other package in Spack.

File: var/spack/repos/builtin/packages/cns-nospec/package.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    variant('mkverbose', default=True, description='Build with verbose')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cns-nospec/package.py]: What does this mean? Does this just make the build more verbose? We should probably always do this. We don't want `cns-nospec+mkverbose` and `cns-nospec~mkverbose` to be different installations with different hashes if they both install the same thing.

File: var/spack/repos/builtin/packages/cns-nospec/package.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    variant('omp', default=False, description='Build with OpenMP support')
+    variant('prof', default=False, description='Build with profiling')
+
+    @property
+    def build_directory(self):
+        return join_path(self.stage.source_path, 'MiniApps', 'CNS_NoSpec')
+
+    def edit(self, spec, prefix):
+        makefile = FileFilter('GNUmakefile')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cns-nospec/package.py]: Does this require a dependency on `gmake`? BSD-make doesn't look for this filename. According to the [GNU make documentation](https://www.gnu.org/software/make/manual/make.html):

> The first name checked, GNUmakefile, is not recommended for most makefiles. You should use this name if you have a makefile that is specific to GNU make, and will not be understood by other versions of make. Other make programs look for makefile and Makefile, but not GNUmakefile.

Comment (jiseung) [inline var/spack/repos/builtin/packages/cns-nospec/package.py]: I'm not entirely sure. However, when I try to build with the updated package with `depends_on(gmake, type='build')`, it encounters a process error in `build()`. If I take that line out, it builds successfully.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cns-nospec/package.py]: `ProcessError` just means something went wrong, not very descriptive. Can you post the contents of the build log?

File: var/spack/repos/builtin/packages/cns-nospec/package.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    variant('mpi', default=True, description='Build with MPI support')
+    variant('debug', default=False, description='Build with debugging')
+    variant('omp', default=False, description='Build with OpenMP support')
+    variant('prof', default=False, description='Build with profiling')
+
+    depends_on('mpi', when='+mpi')
+
+    @property
+    def build_directory(self):
+        return join_path(self.stage.source_path, 'MiniApps', 'CNS_NoSpec')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cns-nospec/package.py]: You can replace this whole thing with:
```python
build_directory = 'MiniApps/CNS_NoSpec'
```

File: var/spack/repos/builtin/packages/cns-nospec/package.py (hunk 4 above)
Comment (jiseung) [inline var/spack/repos/builtin/packages/cns-nospec/package.py]: Actually, I was wrong. It was an IO error of not being able to find the GNUmakefile. 

```[jiseung@jlselogin1 CNS_Nospec]$ spack install cns-nospec~mpi
==> gmake is already installed in /home/jiseung/spack/opt/spack/linux-rhel7-x86_64/gcc-4.8.5/gmake-4.0-mq4dfnxz7enoadmikn3vkvupxfgk3uzo
==> Installing cns-nospec
==> Using cached archive: /home/jiseung/spack/var/spack/cache/cns-nospec/cns-nospec-master.tgz
==> Already staged cns-nospec-master-n4wzgjsugoz4l5k2go4ex65lttgriw7a in /home/jiseung/spack/var/spack/stage/cns-nospec-master-n4wzgjsugoz4l5k2go4ex65lttgriw7a
==> Already patched cns-nospec
==> Building cns-nospec [MakefilePackage]
==> Executing phase : 'edit'
==> Error: IOError: [Errno 2] No such file or directory: 'GNUmakefile'
/home/jiseung/spack/var/spack/repos/proxy-apps-spack/packages/cns-nospec/package.py:56, in edit:
     51       def edit(self, spec, prefix):
     52           makefile = FileFilter('GNUmakefile')
     53           if '+mpi' in spec:
     54               makefile.filter('MPI .*', 'MPI := t')
     55           if '+debug' not in spec:
  >> 56               makefile.filter('NDEBUG.*', '#')
     57           if '+omp' in spec:
     58               makefile.filter('OMP.*', 'OMP := t')
     59           if '+prof' in spec:
     60               makefile.filter('PROF.*', 'PROF := t')

```

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cns-nospec/package.py]: Yeah, you need to be in the correct working directory:
```python
with working_dir(self.build_directory):
```

Comment (jiseung) [inline var/spack/repos/builtin/packages/cns-nospec/package.py]: It seems to be something else. I tried it again with `depends_on('gmake'...)` commented out, and it's still giving me the same error

**Pull request merged**